### PR TITLE
[qa] Add wallet-hd test

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

File: qa/rpc-tests/wallet-hd.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        # Assert that derivation is deterministic
+        hd_add_2 = None
+        for _ in range(num_hd_adds):
+            hd_add_2 = self.nodes[1].getnewaddress()
+        assert_equal(hd_add, hd_add_2)
+
+        # Needs rescan?
+        self.stop_node(1)
+        self.nodes[1] = start_node(1, self.options.tmpdir, self.node_args[1].extend(['-rescan']))
```
Comment (jonasschnelli) [inline qa/rpc-tests/wallet-hd.py]: This won't trigger a rescan. It should be `self.node_args[1] + ['-rescan']`.
`self.node_args[1].extend(['-rescan'])` results in `null?.

Comment (MarcoFalke) [inline qa/rpc-tests/wallet-hd.py]: Awesome! It is always good to have fresh eyes look on the code.

Will push a fix and then squash the commits.

**Comment by jonasschnelli** [comment]
Thanks for the test!
Tested ACK fade505e8b0fa37e2d99bf94259250271a9604d4

**Comment by laanwj** [comment]
Thanks for the test @MarcoFalke and thanks for investigating the issue so quickly during the meeting yesterday @jonasschnelli 
ACK fade505

**Pull request merged**